miden/tests/integration/air/memory.rs

```diff
@@ -21,15 +22,46 @@ fn pop() {
     let asm_op = "pop.mem.0";
     let pub_inputs = vec![1];
 
-    build_op_test!(asm_op, &pub_inputs).prove_and_verify(pub_inputs, 0, false);
+    build_op_test!(asm_op, &pub_inputs).prove_and_verify(pub_inputs.clone(), 0, false);
+}
+
+#[test]
+fn helper_pop() {
+    let asm_op = "begin pop.mem.0 pop.mem.0 pop.mem.0 end";
+    let pub_inputs = vec![1, 2];
+
+    let exe_trace = build_test!(asm_op, &pub_inputs).execute().unwrap();
+    // Sequence of operations: [Span, Pad, MStore, Drop, Pad, Mstore, Drop, Pad, Mstore, Drop]
+    let helper_regs1 = [2, 0, 0, 0, 0, 0].to_elements();
```

Comment:
minor nit: is there a reason to have "1" at the end of "helper_regs"? Since you are redefining the variable later, maybe just leave it as "helper_regs"?